pennylane/resource/resource.py

```diff
@@ -0,0 +1,169 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+The data class which will aggregate all the resource information from a quantum workflow.
+"""
+
+
+class Resources:
```

Comment:
What is the motivation to not use `@dataclass`? Is `Resources` too complex in the end for a simple `dataclass` case?

Reply:
Not really, we could use the data class decorator, I just found that there wasn't any benefit since we override the `__init__` method anyways. I guess we could save ~15 lines of code if we use the decorator since we don't have to implement `__repr__` and `__eq__`. 

I can make the change if you think it makes more sense, in my opinion it's just a matter of preference at this point.